bootstrap.sh

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env bash
+
+# Usage instructions for cloning and setting up sparse-checkout
+# 1. Clone the repository sparsely:
+#    git clone --filter=blob:none --sparse https://github.com/freeCodeCamp/freeCodeCamp.git
```

Comment:
This helps me a lot.

```suggestion
#    git clone --depth=1 --filter=blob:none --sparse https://github.com/freeCodeCamp/freeCodeCamp.git
```

Reply:
The issue with shallow clones is it is great for CI/CD (even one-off branches like on Gitpod and Codespaces) but rarely works in a localized setup.

In a local setup, you are very likely to switch branches, rebase, etc. which become buggy or non-functional at all.

I am aiming for only the first time setup (with incremental get me more files) for now.